google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/scaladsl/spray/BigQueryRestJsonProtocol.scala

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.googlecloud.bigquery.scaladsl.spray
+
+import spray.json.{AdditionalFormats, CollectionFormats, ProductFormats, StandardFormats}
+
+/**
+ * Provides all the predefined JsonFormats for the BigQuery REST API.
+ */
+trait BigQueryRestJsonProtocol
```

Comment:
There are lots of datatypes to support. I noticed that the date/datetime/timestamp related fields don't have a formats representation. Should we add those?

Are all the provided formats under test?

Reply:
There are! I noticed a new preview datatype `BIGNUMERIC` as well.

I can definitely add the date/time stuff, I think I avoided this because people tend to prefer 3rd-party libs for this? E.g. Joda time.

But yes, all the formats should be covered by the [end-to-end test](https://github.com/armanbilge/alpakka/blob/wip-new-bigquery/google-cloud-bigquery/src/test/scala/akka/stream/alpakka/googlecloud/bigquery/e2e/A.scala).

Reply:
Decided to add these! They are covered by the E2E test.

Reply:
I just added support+tests for the `BYTES` type via Akka's `ByteString`. So that covers about everything, besides the `GEOGRAPHY` type.